src/config.rs

```diff
@@ -1000,6 +993,15 @@ fn get_default_host_triple(s: &Settings, process: &Process) -> dist::TargetTripl
         .unwrap_or_else(|| dist::TargetTriple::from_host_or_build(process))
 }
 
+fn non_empty_var(name: &str, process: &Process) -> anyhow::Result<Option<String>> {
```

Comment:
Can we make this `Process::non_empty_var` instead?

Reply:
I guess we could? Given that there are only two callers and both of them are in `config`, I'd rather keep it private to the module (unless you think there are other callers that could benefit).

Reply:
@djc Yup, keeping the `impl Process` block local to this module would be perfect!

Reply:
Hmm, not really a fan of keeping `impl` blocks (with the same bounds) in separate files... What's wrong with the free function setup?

Reply:
@djc It's just that a free `non_empty_var` function makes me feel less sure about what the function is doing without having to look at the implementation. Not a hard blocker really.

Reply:
Would renaming it to `non_empty_env_var()` (or some other name) help with that?

Reply:
> Would renaming it to `non_empty_env_var()` (or some other name) help with that?

@djc Yes, I think that would be fine as well. Thanks!

Reply:
Done!